tensorflow/python/ops/metrics_impl.py

```diff
@@ -263,8 +263,9 @@ def mean(values, weights=None, metrics_collections=None,
       values = math_ops.multiply(values, weights)
       num_values = math_ops.reduce_sum(weights)
 
-    update_total_op = state_ops.assign_add(total, math_ops.reduce_sum(values))
-    update_count_op = state_ops.assign_add(count, num_values)
+    with ops.control_dependencies([values]):
+      update_total_op = state_ops.assign_add(total, math_ops.reduce_sum(values))
```

Comment:
So technically, this doesn't ened to be inside the `control_dependencies`, right?